### PR TITLE
[data-export] Input focus on SOQL query field

### PR DESCRIPTION
**Pull request by asknet**
Problem statement:
Whenever I open data export, the query text area isn't focused. It requires another click to set focus and then start typing.

Solution:
This PR allows to set query textarea as focus on load.

Also new to reactjs development. Please review and suggest.

### CODE REVIEW

**Comment by tprouvot** [comment]
Hi @asknet 
Thanks for this PR ! 

Could you add the new feature description in CHANGES.md file ? (following current convention)

Could you condition this focus based on a localStorage variable (by default false) ?
Since the auto focus triggers the api call to query the object definition, I would prefer to let users decide if they want to enable this or not ?

We should have an option page where users will be able to enable / disable features from this page.
Waiting for this the configuration can be made from chrome dev console.

Thanks a lot !

**Comment by asknet** [comment]
@tprouvot 

Thanks for the feedback.

Sure, I will update the `CHANGES.md` file.
IMO, by default this should be enabled and can be turned off via setting as suggested. Whenever a data export is launched, I'm under the opinion that user wants to type some query. The only exception is they might choose a query from history or saved ones. Even in such cases, making an extra api call for object definition on focus shouldn't pose a concern.

**Comment by tprouvot** [comment]
@asknet 
Yes most of the time they will use this page to run a query but not necessarily on the last object queried.
I wouldn't mind to set the autofocus by default to true if the API call is not made.

**Comment by toly11** [comment]
I came in here to create a pull request for this exact feature myself:)
Great addition, please merge this.

File: addon/data-export.js [hunk 1]
```diff
@@ -1049,6 +1049,8 @@ class App extends React.Component {
     let queryInput = this.refs.query;
 
     model.setQueryInput(queryInput);
+    //Set the cursor focus on query text area
+    queryInput.focus();
```
Comment (tprouvot) [inline addon/data-export.js:1053]:
```suggestion
    //Set the cursor focus on query text area
    if (localStorage.getItem("disableQueryInputAutoFocus") !== "true"){
      queryInput.focus();
    }
```

**Review by tprouvot — CHANGES_REQUESTED** [review]
Hi @asknet,
I changed my mind, we can enable this param by default and it can be used to disable it.
 Could you accept the suggested changes and also update the CHANGES.md file ? 
Thank you !

**Comment by asknet** [comment]
@tprouvot Thank you. I updated it. Kindly, review.

File: CHANGES.md [hunk 2]
```diff
@@ -18,6 +18,7 @@
 - Update pop-up release note link to github pages
 - Detect SObject on listview page [feature 121](https://github.com/tprouvot/Salesforce-Inspector-reloaded/issues/121) (idea by [Mehdi Cherfaoui](https://github.com/mehdisfdc))
 - Automate test setup manual step of contact to multiple accounts [Aidan Majewski](https://github.com/aimaj)
+- In Data export, set input focus in SQOL query text area. [PR153](https://github.com/tprouvot/Salesforce-Inspector-reloaded/pull/153)
```
Comment (tprouvot) [inline CHANGES.md:21]:
```suggestion
- In Data export, set input focus in SQOL query text area. [feature 183](https://github.com/tprouvot/Salesforce-Inspector-reloaded/issues/183) (contribution by [Sarath Addanki](https://github.com/asknet))
```

**Comment by tprouvot** [comment]
#153

**Comment by tprouvot** [comment]
Option added to Data Export tab option
image

**Pull request merged**